### ISSUE TITLE
PyPy support

### ISSUE THREAD

**Issue by oberstet** [pymssql/pymssql#408]
It would be nice to be able to use pymssql under PyPy.

This is what I get:

```console
oberstet@bvr-sql18:~$ sudo /opt/pypy400/bin/pip install pymssql
The directory '/home/oberstet/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/oberstet/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting pymssql
  Downloading pymssql-2.1.2.tar.gz (898kB)
    100% |████████████████████████████████| 901kB 62kB/s
Installing collected packages: pymssql
  Running setup.py install for pymssql ... error
    Complete output from command /opt/pypy400/bin/pypy -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-5AceR8/pymssql/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-wNgN0s-record/install-record.txt --single-version-externally-managed --compile:
    setup.py: platform.system() => 'Linux'
    setup.py: platform.architecture() => ('64bit', 'ELF')
    setup.py: platform.linux_distribution() => ('debian', 'jessie/sid', '')
    setup.py: platform.libc_ver() => ('glibc', '2.2.5')
    setup.py: Not using bundled FreeTDS
    setup.py: include_dirs = ['/usr/local/include']
    setup.py: library_dirs = ['/usr/local/lib']
    running install
    running build
    running build_ext
    building '_mssql' extension
    creating build
    creating build/temp.linux-x86_64-2.7
    cc -O2 -fPIC -Wimplicit -I/usr/local/include -I/opt/pypy400/include -c _mssql.c -o build/temp.linux-x86_64-2.7/_mssql.o -DMSDBLIB
    _mssql.c:692:3: error: unknown type name ‘PyBaseExceptionObject’
       PyBaseExceptionObject __pyx_base;
       ^
    _mssql.c: In function ‘__pyx_f_6_mssql__quote_simple_value’:
    _mssql.c:19366:3: warning: implicit declaration of function ‘PyByteArray_Check’ [-Wimplicit-function-declaration]
       __pyx_t_7 = PyByteArray_Check(__pyx_v_value);
       ^
    _mssql.c: In function ‘init_mssql’:
    _mssql.c:24368:99: error: ‘PyBaseExceptionObject’ undeclared (first use in this function)
       __pyx_ptype_6_mssql_Exception = __Pyx_ImportType(__Pyx_BUILTIN_MODULE_NAME, "Exception", sizeof(PyBaseExceptionObject), 0); if (unlikely(!__pyx_ptype_6_mssql_Exception)) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 143; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
                                                                                                       ^
    _mssql.c:24368:99: note: each undeclared identifier is reported only once for each function it appears in
    error: command 'cc' failed with exit status 1

    ----------------------------------------
Command "/opt/pypy400/bin/pypy -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-5AceR8/pymssql/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-wNgN0s-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-5AceR8/pymssql
oberstet@bvr-sql18:~$
```

I'm note sure if this is fixable at all. The use of Cython is a problem. However, Cython has a [porting guide](http://docs.cython.org/src/userguide/pypy.html).

> Note: under PyPy (and also CPython), interfacing with native code, the "recommended way" is [cffi](https://cffi.readthedocs.org/en/latest/) nowerdays.

**Comment by ramiro** [comment]
Duplicate of #309
